examples/flutter_gallery/lib/gallery/app.dart

```diff
@@ -157,21 +157,27 @@ class GalleryAppState extends State<GalleryApp> {
       };
     }
 
-    return new MaterialApp(
-      title: 'Flutter Gallery',
-      color: Colors.grey,
-      theme: _galleryTheme.theme.copyWith(platform: _platform ?? defaultTargetPlatform),
-      showPerformanceOverlay: _showPerformanceOverlay,
-      checkerboardRasterCacheImages: _checkerboardRasterCacheImages,
-      checkerboardOffscreenLayers: _checkerboardOffscreenLayers,
-      routes: _kRoutes,
-      home: home,
-      builder: (BuildContext context, Widget child) {
-        return new Directionality(
-          textDirection: _overrideDirection,
-          child: _applyScaleFactor(child),
-        );
-      },
+    return new Semantics(
+      textDirection: _overrideDirection,
+      child: new MaterialApp(
+        title: 'Flutter Gallery',
+        color: Colors.grey,
+        theme: _galleryTheme.theme.copyWith(platform: _platform ?? defaultTargetPlatform),
+        showPerformanceOverlay: _showPerformanceOverlay,
+        checkerboardRasterCacheImages: _checkerboardRasterCacheImages,
+        checkerboardOffscreenLayers: _checkerboardOffscreenLayers,
+        routes: _kRoutes,
+        home: home,
+        builder: (BuildContext context, Widget child) {
+          return new Semantics(
+            textDirection: _overrideDirection,
+            child: new Directionality(
```

Comment:
Directionality is not enough to set the semantic textDirection? Might be worth filing an issue about this.

Reply:
Done. https://github.com/flutter/flutter/issues/16835